client/web/compose/src/components/ModuleFields/Editor/Record.vue

```diff
@@ -292,6 +297,20 @@ export default {
       updateRecords: 'record/updateRecords',
     }),
 
+    createEvents () {
+      this.$root.$on('record-field-change', this.refetchOnPrefilterValueChange)
+    },
+
+    refetchOnPrefilterValueChange ({ fieldName }) {
+      if (!fieldName) return
+
+      const { prefilter } = this.field.options
+      if (!(prefilter || '').includes('${record') && !(prefilter || '').includes(fieldName)) return
+
+      this.selected = { value: undefined }
+      this.loadLatest()
```

Comment:
shouldnt you use fetchPrefiltered? Since loadLatest doesnt have the updated filter as far as i can see.